scripts/custom_lint_checks.py

```diff
@@ -19,67 +19,115 @@
     priority = -1
     msgs = {
         'C0001': (
-            'Keyword argument(s) should be named explicitly in function call.',
-            'non-explicit-kwargs',
+            'Keyword argument %s should be named explicitly in %s call of %s.',
+            'non-explicit-keyword-args',
             'All keyword arguments should be explicitly named in function call.'
         ),
     }
 
-    def __init__(self, linter=None):
-        """Constructs an ExplicitKwargsChecker object.
+    def visit_call(self, node):
+        """Visits each function call in a lint check.
 
         Args:
-            linter: Pylinter. An object implementing Pylinter.
+            node. Call. The current function call node.
         """
-        super(ExplicitKwargsChecker, self).__init__(linter)
-        self._function_name = None
-        self._defaults_count = 0
-        self._positional_arguments_count = 0
+        called = checker_utils.safe_infer(node.func)
 
-    def visit_functiondef(self, node):
-        """Visits each function definition in a lint check.
+        try:
+            # For the rationale behind the Pylint pragma below,
+            # see https://stackoverflow.com/a/35701863/8115428
+            called, implicit_args, callable_name = (
+                typecheck._determine_callable(called))  # pylint: disable=protected-access
+        except ValueError:
+            return
 
-        Args:
-            node. FunctionDef. The current function definition node.
-        """
-        self._function_name = node.name
+        if called.args.args is None:
+            # Built-in functions have no argument information.
+            return
 
-    def visit_arguments(self, node):
-        """Visits each function argument in a lint check.
+        if len(called.argnames()) != len(set(called.argnames())):
+            return
 
-        Args:
-            node. Arguments. The current function arguments node.
-        """
-        if node.defaults is not None:
-            self._defaults_count = len(node.defaults)
-            if node.args is not None:
-                self._positional_arguments_count = (
-                    len(node.args) - len(node.defaults))
+        # Build the set of keyword arguments and count the positional arguments.
+        call_site = astroid.arguments.CallSite.from_call(node)
+        if call_site.has_invalid_arguments() or (
+                call_site.has_invalid_keywords()):
+            return
 
-    def visit_call(self, node):
-        """Visits each function call in a lint check.
+        num_positional_args = len(call_site.positional_arguments)
+        keyword_args = list(call_site.keyword_arguments.keys())
+
+        already_filled_positionals = getattr(called, 'filled_positionals', 0)
+        already_filled_keywords = getattr(called, 'filled_keywords', {})
+
+        keyword_args += list(already_filled_keywords)
+        num_positional_args += implicit_args + already_filled_positionals
```

Comment:
Separate the lists into separate extend lines

Reply:
Sorry, I did not quite get you here -- can you please explain what needs to be changed?
Thanks!

Reply:
Got it, done! Thanks!